### PR TITLE
knex cli list all migrations with status (#3390)

### PR DESCRIPTION
**Pull request by vithalreddy**
Add's `list` or `status` command to knex cli to show both completed and pending migrations.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -267,6 +267,47 @@ function invoke(env) {
         .catch(exit);
     });
 
+  commander
+    .command('list')
+    .alias('status')
+    .description('        List all migrations files with status.')
+    .action(() => {
+      pending = initKnex(env, commander.opts())
```
Comment (kibertoad) [inline bin/cli.js]: can this use async/await instead of promise chain?

File: bin/cli.js [hunk 2]
```diff
@@ -267,6 +267,47 @@ function invoke(env) {
...
+        .migrate.list()
+        .then(([completed, newMigrations]) => {
```
Comment (kibertoad) [inline bin/cli.js]: Would be great if you could extract whole logic for this operation into a separate file and only call it from cli.js

File: lib/migrate/Migrator.js [hunk 3]
```diff
@@ -219,6 +219,27 @@ class Migrator {
       });
   }
 
+  // list all migrations
+  list(config) {
+    this._disableProcessing();
+    this.config = getMergedConfig(config, this.config);
+
+    return migrationListResolver
```
Comment (kibertoad) [inline lib/migrate/Migrator.js]: could you please use async/await instead?

File: test/jake/jakelib/migrate-test.js [hunk 4]
```diff
@@ -543,6 +543,98 @@ test('migrate:down undos only the last run migration', (temp) => {
...
+    `node ${KNEX} list \
+    --client=sqlite3 \
+    --connection=${temp}/db \
+    --migrations-directory=${temp}/migrations`
+  )
+    .then(({ stdout }) => {
+      assert.include(stdout, `No Completed Migration files Found.`);
+      assert.include(stdout, `No Pending Migration files Found.`);
+
+      fs.writeFileSync(
```
Comment (kibertoad) [inline test/jake/jakelib/migrate-test.js]: using FileHelper with auto-cleanup is preferable for preparing files. see https://github.com/tgriesser/knex/blob/master/test/cli/migrate-make.spec.js

File: test/jake/jakelib/migrate-test.js [hunk 5]
```diff
@@ -543,6 +543,98 @@ test('migrate:down undos only the last run migration', (temp) => {
...
+
+      return assertExec(
+        `node ${KNEX} migrate:up \
+        --client=sqlite3 \
+        --connection=${temp}/db \
+        --migrations-directory=${temp}/migrations`,
+        'create_books_table'
+      );
+    })
+    .then(({ stdout }) => {
```
Comment (kibertoad) [inline test/jake/jakelib/migrate-test.js]: async/await is preferable

**Comment by kibertoad** [comment]
Much appreciated! Can you also create a documentation PR in https://github.com/knex/documentation?

**Comment by vithalreddy** [comment]
@kibertoad Thanks for the review, Will do the required changes.

File: bin/utils/cli-config-utils.js [hunk 6]
```diff
@@ -95,6 +95,38 @@ function success(text) {
   process.exit(0);
 }
 
+function print_migrations_list(completed, newMigrations) {
```
Comment (kibertoad) [inline bin/utils/cli-config-utils.js]: I wouldn't say that config-utils is a good place for this particular set of logic, I would create a new file "migrationsLister". Also please use camelCase for function name ;)

File: test/jake/jakelib/migrate-test.js [hunk 7]
```diff
@@ -543,6 +543,100 @@ test('migrate:down undos only the last run migration', (temp) => {
     });
 });
 
+test('list prints migrations both completed and pending', async (temp) => {
+  const migrationFile1 = '001_create_animals_table.js';
+  const migrationFile2 = '002_add_age_column_to_animals_table.js';
+
+  let { stdout } = await assertExec(
```
Comment (kibertoad) [inline test/jake/jakelib/migrate-test.js]: does this need to be `let`? I assume we don't overwrite it anywhere.

Comment (vithalreddy) [reply test/jake/jakelib/migrate-test.js]: It is getting used below as well.

File: bin/utils/migrationsLister.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+const color = require('colorette');
+
+const { success } = require('./cli-config-utils');
+
+function migrationsLister(completed, newMigrations) {
```
Comment (kibertoad) [inline bin/utils/migrationsLister.js]: Sorry for the constant stream of requests. Could you please rename function to be a verb - "listMigrations"? File name is fine.

Comment (vithalreddy) [reply bin/utils/migrationsLister.js]: No Problem at all.
Will rename and push the changes.

File: test/jake/jakelib/migrate-test.js (hunk 7 above)
Comment (kibertoad) [reply test/jake/jakelib/migrate-test.js]: Can we just have to separate instances of stdout then? Reassigning variables in this fashion is confusing and is better avoided.

**Comment by kibertoad** [comment]
@vithalreddy Thanks for doing the changes! This can be merged after documentation is created.

**Comment by vithalreddy** [comment]
Thanks. Will add the documentation and submit PR for the same.

File: bin/cli.js [hunk 9]
```diff
@@ -267,6 +268,19 @@ function invoke(env) {
         .catch(exit);
     });
 
+  commander
+    .command('list')
```
Comment (kibertoad) [inline bin/cli.js]: can you change it to be `migrate:list` for consistency?

File: bin/cli.js [hunk 10]
```diff
@@ -267,6 +268,19 @@ function invoke(env) {
...
+    .alias('status')
```
Comment (kibertoad) [inline bin/cli.js]: migrate:status as alias

**Comment by kibertoad** [comment]
Thanks!

**Comment by vithalreddy** [comment]
@kibertoad Thanks for accepting and reviewing my pr.

**Pull request merged**